derive_client/clients/base_client.py

```diff
@@ -781,3 +787,280 @@ def transfer_from_subaccount_to_funding(self, amount: int, asset_name: str, suba
+
+        # Create taker action (recipient)
+        taker_action = SignedAction(
+            subaccount_id=to_subaccount_id,
+            owner=self.wallet,
+            signer=self.signer.address,
+            signature_expiry_sec=MAX_INT_32,
+            nonce=get_action_nonce(),  # taker_nonce
+            module_address=self.config.contracts.TRADE_MODULE,
+            module_data=TakerTransferPositionModuleData(
+                asset_address=instrument["base_asset_address"],
+                sub_id=int(instrument["base_asset_sub_id"]),
+                limit_price=transfer_price,
+                amount=transfer_amount,
+                recipient_id=to_subaccount_id,
+                position_amount=original_position_amount,
+            ),
+            DOMAIN_SEPARATOR=self.config.DOMAIN_SEPARATOR,
+            ACTION_TYPEHASH=self.config.ACTION_TYPEHASH,
+        )
+
+        # Sign both actions
+        maker_action.sign(self.signer.key)
+        taker_action.sign(self.signer.key)
+
+        # Create request parameters
+        maker_params = {
+            "direction": maker_action.module_data.get_direction(),
+            "instrument_name": instrument_name,
+            **maker_action.to_json(),
+        }
+
+        taker_params = {
+            "direction": taker_action.module_data.get_direction(),
+            "instrument_name": instrument_name,
+            **taker_action.to_json(),
+        }
+
+        payload = {
+            "wallet": self.wallet,
+            "maker_params": maker_params,
+            "taker_params": taker_params,
+        }
+
+        response_data = self._send_request(url, json=payload)
+
+        # Extract transaction_id from response for polling
+        transaction_id = self._extract_transaction_id(response_data)
+        return wait_until(
+            self.get_transaction,
+            condition=_is_final_tx,
+            transaction_id=transaction_id,
+        )
+
+    def get_position_amount(self, instrument_name: str, subaccount_id: int) -> float:
+        """
+        Get the current position amount for a specific instrument in a subaccount.
+
+        This is a helper method for getting position amounts to use with transfer_position().
+
+        Parameters:
+            instrument_name (str): The name of the instrument.
+            subaccount_id (int): The subaccount ID to check.
+
+        Returns:
+            float: The current position amount.
+
+        Raises:
+            ValueError: If no position found for the instrument in the subaccount.
+        """
+        positions = self.get_positions()
+        for pos in positions.get("positions", []):
+            if pos["instrument_name"] == instrument_name and pos["subaccount_id"] == subaccount_id:
+                return float(pos["amount"])
+
+        raise ValueError(f"No position found for {instrument_name} in subaccount {subaccount_id}")
+
+    def transfer_positions(
+        self,
+        positions: list[TransferPosition],
```

Comment:
Transfer position should use this as well rather than replicating the logic youve alredy made an object to handle